### PR TITLE
New mission: Combat

### PR DESCRIPTION
**Pull request by robothauler**
The player has to jump to an unpopulated or lawless system to do the job. Additional fuel is required to finish the mission in time. So a balanced ship between speed, firepower and protection is needed.

### CODE REVIEW

**Comment by WKFO** [comment]
Good idea, descriptions are detailed and mission is pretty enjoyable too as you have to make some managements of ship equipment and combat-spaceflight problem solving. (Enemies can often be very powerful, also you need to balance the time you spend on casual spaceflight phase.)

But, very often, I arrive at the destination body and orbit it; then a few in-game minutes later I get the message "Primary objective completed." without encountering with anything at all (not the peaceful recon mission though, the full-on assault one). I suspect the enemy crashes into the celestial body or sth while trying to intercept me.

This is not a problem of the mission, but rather a problem of the combat AI itself. When the AI tries to intercept the player it doesn't even slightly care about crashing into ground or other objects as far as I've observed.

Anyway, pretty nice mission. Would love to see this in the game.

**Comment by robothauler** [comment]
@WKFO nice video! 

The missions were very cheap. So I doubled the payment.

The names for the police and military forces are taken from the factions in data/factions/.
If you have started a military mission at a station in the Commonwealth of Independent Worlds territory then you have to go back to them to get payed.

File: data/lang/module-combat/en.json [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+      "description" : "{area} is an asteroid or planet",
+      "message" : "In response to a case of piracy in the {system} system ({sectorx}, {sectory}, {sectorz}), {dist} ly, the local authorities pay {cash} for a combat mission at {area}."
+   },
+   "MILITARY_3" : {
+      "description" : "",
+      "message" : "The counter intelligence corps needs a civil pilot who asks no questions for a {mission} mission in the {system} system ({sectorx}, {sectory}, {sectorz}), {dist} ly. Pay is {cash}."
+   },
+   "MILITARY_IT_MUST_BE_COMPLETED_BY" : {
+      "description" : "",
+      "message" : "A task force will be assembled to regain control of {area}. So it's important that the mission is successful completed by {date}."
```
Comment (zonkmachine) [inline data/lang/module-combat/en.json]: Nip/Tuck...

> So it's important that the mission is successful completed by {date}."

**It's** important that the mission is **successfully** completed by {date}."

File: data/lang/module-combat/en.json [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+      "description" : "",
+      "message" : "It's important that the mission is completed by {date}, because more hostile ships are on there way."
+   },
+   "POLICE_LAND_THERE" : {
+      "description" : "",
+      "message" : "Just land at a station with a {org} office."
+   },
+   "HARVESTING_1" : {
+      "description" : "{area} is an asteroid or planet",
+      "message" : "The {org} will pay {cash} for a {mission} mission in the {system} system ({sectorx}, {sectory}, {sectorz}). We have reason to believe that a ruthless market competitor is responsible for the disapperance of our freighters at {area}. Distance to the target area is {dist} ly."
```
Comment (zonkmachine) [inline data/lang/module-combat/en.json]: > disapperance

disappearance

File: data/lang/module-combat/en.json [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+      "description" : "{area} is an asteroid or planet",
+      "message" : "Our market competitor is seriously claiming he has ownership of {area} in the {system} system ({sectorx}, {sectory}, {sectorz}). The {org} totally disagree and is willing to pay {cash} for a combat mission at {area}. {area} is in a distance of {dist} ly."
+   },
+   "HARVESTING_3" : {
+      "description" : "{area} is an asteroid or planet",
+      "message" : "Please help! We received a distress call from one of our freighters orbiting {area} in the {system} system ({sectorx}, {sectory}, {sectorz}). The call was interrupted, so we fear the worst. Distance to {system} is {dist} ly. The pay is {cash}."
+   },
+   "HARVESTING_IT_MUST_BE_COMPLETED_BY" : {
+      "description" : "",
+      "message" : "It's important that the mission is completed by {date}. Please keep shutm about this operation."
```
Comment (zonkmachine) [inline data/lang/module-combat/en.json]: > shutm

shtum/schtum?

Comment (Zireael07) [reply data/lang/module-combat/en.json]: He probably meant keep shut, I would change to keep quiet

File: data/lang/module-combat/en.json [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+      "message" : "Police detectives have discovered that a group of pirates try to establish a stronghold on {area} in the {system} system ({sectorx}, {sectory}, {sectorz}), {dist} ly. This must not happen. The local authorities pay {cash} for a mission to {area} as preparation of a police operation."
+   },
+   "POLICE_2" : {
+      "description" : "{area} is an asteroid or planet",
+      "message" : "The {org} will pay {cash} for a combat mission in the {system} system ({sectorx}, {sectory}, {sectorz}). We are short on pilots and ships. So this will be an opportunity for a loyal citizen to make money. The target is a small pirate base on {area}. You need hyperdrive propellant for {dist} ly and additional fuel for the return route."
+   },
+   "POLICE_3" : {
+      "description" : "{area} is an asteroid or planet",
+      "message" : "Our markets were flooded by contrabant. The source has been traced back to a hidden base at {area} in the {system} system ({sectorx}, {sectory}, {sectorz}). Only a distance of {dist} ly. Because of the reduced police budget the local authorities would like to outsource this mission and have allotted {cash} credits for this operation."
```
Comment (zonkmachine) [inline data/lang/module-combat/en.json]: > contrabant

contraband

**Comment by robothauler** [comment]
Thank you for the review. I have changed the text.

**Comment by impaktor** [comment]
Sorry this is stuck, but I blame I'm late with my tax return :wink: and other stuff.

In the mean time, if any more players care to play test it, then fine.

If it's not merged by Christmas, I'll smash that merge button anyway... (I'm half joking, but probability isn't zero for it taking that long, it seems).

**Comment by nozmajner** [comment]
@robothauler 
Here are two possible icons for the combat missions:
screenshot-20180703-183637
[The upper one's a little fighter shooting](https://i.imgur.com/yo7GJSv.png)
[The lower one's the light fighter icon from the ship classes](https://i.imgur.com/HLAjxr8.png), so it's a bit more abstarct you could say.
(And one for the racing mission in the middle). 
Ready to use pngs under the URLs

**Comment by robothauler** [comment]
@nozmajner Great! The lower one's looks good to me. Thank you.

**Pull request merged**